pkg/apis/resource/types.go

```diff
@@ -479,20 +480,22 @@ type ResourceClaimTemplateList struct {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// NodeResourceSlice provides information about available
+// ResourceSlice provides information about available
 // resources on individual nodes.
-type NodeResourceSlice struct {
+type ResourceSlice struct {
 	metav1.TypeMeta
 	// Standard object metadata
 	metav1.ObjectMeta
 
-	// NodeName identifies the node where the capacity is available.
-	// A field selector can be used to list only NodeResourceSlice
+	// NodeName identifies the node which provides the resources
```

Comment:
I'm not going to argue it now, but I think for .31 we should consider how a non-node resource would be represented.  For example, should it be a selector?